internal/npm_install/generate_build_file.js

```diff
@@ -317,15 +344,57 @@ function flattenDependencies(pkg, dep, pkgsMap) {
  */
 function printJson(pkg) {
   // Clone and modify _dependencies to avoid circular issues when JSONifying
+  // & delete _files array
   const cloned = {...pkg};
   cloned._dependencies = cloned._dependencies.map(dep => dep._dir);
+  delete cloned._files;
   return JSON.stringify(cloned, null, 2).split('\n').map(line => `# ${line}`).join('\n');
 }
 
+/**
+ * A filter function for a bazel filegroup.
+ * @param files array of files to filter
+ * @param exts list of white listed extensions; if empty, no filter is done on extensions;
+ *             '' empty string denotes to allow files with no extensions, other extensions
+ *             are listed with '.ext' notation such as '.d.ts'.
+ */
+function filterFilesForFilegroup(files, exts = []) {
+  files = files
+              // Bazel target names must be composed entirely of characters drawn
+              // from the set a–z, A–Z, 0–9, and the punctuation symbols _/.+-=,@~.$
+              // Files with spaces are also not allowed in Bazel runfiles
```

Comment:
"and Bazel doesn't support unicode characters in file names"

Reply:
👍